### PR TITLE
Overhaul automated review CI.

### PR DESCRIPTION
**Pull request by Ferroin**
Equivalent to https://github.com/netdata/netdata/pull/11553, as well as adding a few extra linting types that should be here but were missing.

### CODE REVIEW

File: .github/workflows/reviewdog.yml [hunk 1]
```diff
@@ -1,8 +1,90 @@
-name: reviewdog
-on: [pull_request]
+---
+# Runs various ReviewDog based checks against PR with suggested changes to improve quality
+name: Review
+on:
+  pull_request:
+env:
+  DO_NOT_TRACK: 1
```
Comment (ilyam8) [inline .github/workflows/reviewdog.yml:7]: not needed

File: .github/workflows/reviewdog.yml [hunk 2]
```diff
@@ -1,8 +1,90 @@
...
+      golangci-lint: ${{ steps.golangci-lint.outputs.run }}
+      hadolint: ${{ steps.hadolint.outputs.run }}
+      shellcheck: ${{ steps.shellcheck.outputs.run }}
+      yamllint: ${{ steps.yamllint.outputs.run }}
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          fetch-depth: 0
```
Comment (ilyam8) [inline .github/workflows/reviewdog.yml:23]: not needed

Comment (Ferroin) [reply .github/workflows/reviewdog.yml:23]: Just the fetch-depth part, or both that and the submodules part as well?

Comment (ilyam8) [reply .github/workflows/reviewdog.yml:23]: Both. This repo has no submodules.

**Review by ilyam8 — APPROVED** [review]

**Comment by ilyam8** [comment]
Just noticed that all the "Prepare Review Job" steps fail with

> fatal: ambiguous argument 'origin/master': unknown revision or path not in the working tree.

**Comment by ilyam8** [comment]
my bad, it is me who suggested removing "width: fetch-depth: 0"

**Pull request merged**